docs/proposals/backup-restore.md

```diff
@@ -1,131 +1,207 @@
-Each artifact in the output OCI layout will be tagged with the name of source.
-For example, the `registry.k8s.io/kube-apiserver-arm64:v1.31.0` artifact will be tagged `registry.k8s.io/kube-apiserver-arm64:v1.31.0`.
+The desired end-to-end user experience of using `oras backup` and `oras restore` to address the identified problems and support the outlined user scenarios is illustrated below.
+
+#### Backup an Entire Repository to a Tarball and Restore to Another Registry
+
+Assume two tags `v1` and `v2` are stored in a repository `registry.k8s.io/kube-apiserver` and each tag has one referrer. Backup the entire repo to a tarball and restore it to another registry:
 
 ```bash
-% oras repo tags --oci-layout ./mirror
-registry.k8s.io/kube-apiserver-arm64:v1.31.0
-registry.k8s.io/kube-apiserver-arm64:v1.32.0
-registry.k8s.io/kube-controller-manager-arm64:v1.31.0
+## Backup a repository from a registry to a local compressed tarball. All tags and their referrers will be included.
+oras backup --output backup.tar --include-referrers registry-a.k8s.io/kube-apiserver
 ```
 
-The backup command will also have the ability to write output to a new compressed tar file where the contents are in a single oci-layout
-format. For example:
+Upon success, the output will be:
 
-```bash
-oras backup --output ./mirror.tgz  registry.k8s.io/kube-apiserver-arm64:v1.31.0 registry.k8s.io/kube-controller-manager-arm64:v1.31.0
+```console
+Found 2 tag(s) in registry-a.k8s.io/kube-apiserver
+- v1
+- v2
+<progress bar>...
+Exporting to backup.tar
+Successfully backed up 2 tag(s) from registry-a.k8s.io/kube-apiserver to backup.tar
+```
+
+Transfer the backup file to new environment via secure channels (e.g., BitLocker-enabled removable drives).
+
+Restore images and referrer artifacts from a local backup file to a target registry. All tags and their referrers will be included by default.
+
+```console
+oras restore --input backup.tar registry-b.k8s.io/kube-apiserver
 ```
 
-If the output specified is an existing directory, the output will be written in that directory in OCI layout format.
-If the output specified is an existing file, it will be overwritten.
-If the output specified is neither a file or a directory, file output is assumed.
-The file name does NOT need to end in `tgz`, but file output will be compressed tar file.
+Upon success, the output will be:
 
-### oras restore
+```console
+Found 2 tag(s) in backup.tar
+- v1
+- v2
+<progress bar>...
+Pushing to registry-b.k8s.io/kube-apiserver
+Successfully restored 2 tag(s) to registry-a.k8s.io/kube-apiserver
+```
 
-The restore command will support reading a directory or a compressed tar file and writing the content to a remote registry.
+List all tags from the repo `registry-b.k8s.io/kube-apiserver`:
+
+```console
+$ oras repo tags registry-b.k8s.io/kube-apiserver
+v1
+v2
+```
 
-#### Restore from a directory
+#### Backup and Restore an Image with Referrers 
```

Comment:
Would it be better to demonstrate the scenario of backing up multiple artifacts by specifying tags? Such as 

`oras backup registry-a.k8s.io/kube-apiserver:v1,v2 --include-referrers --output airgap-snapshot.tar`

Reply:
Good call out. I added the example of backing up multiple artifacts by specifying tags in the section Backup an Entire Repository to a Tarball and Restore to Another Registry.